Add angle() function to utils.lua

hemmond opened the pull request:
Complementary function for `angleToVector()`, with simillar API as `distance(a,b,c,d)` function, `angle()` returns bearing between first object/position and second object/position.

daid commented:
Due to EE's coordinate system being screwy, it's for me unclear if this function returns a heading or a rotation vector.

Example: `player:setRotation(angle(player, station))` or `player:setHeading(angle(player, station))`?

hemmond commented:
It returns rotation vector (0° is "to the east"). I could rename it to `angleRotation()` and add `angleHeading()` for heading (0° is "to the north"), so the utils.lua provides both functions. I can also try to document this "dual angling" in the documentation of both functions (and rename function arguments to `heading_angle`/`rotation_angle` in `utils.lua` depending on what angle it accepts). Is that OK?

hemmond commented:
So, I have implemented `angleRotation()` and `angleHeading()` instead of original `angle()` function along with its respective documentation comments. I have also introduced `angle_is_heading` optional variable to `setCirclePos()` and `vectorFromAngle()`. It should now provide all functions for both numeric use-cases (rotation angle and heading).
